tempto-core/src/main/java/com/teradata/tempto/internal/convention/ConventionBasedTestProxyGenerator.java

```diff
@@ -81,7 +87,23 @@ public ConventionBasedTest generateProxy(ConventionBasedTest conventionBasedTest
 
     private String generatedClassName(ConventionBasedTest conventionBasedTest)
     {
-        return testPackage + "." + capitalize(replace(conventionBasedTest.testName(), "-", "_"));
+        List<String> testNameParts = Splitter.on('.').splitToList(conventionBasedTest.getTestName());
+        return testPackage + "." + toJavaSymbol(testNameParts.get(testNameParts.size() - 2));
```

Comment:
what is testNameParts.get(testNameParts.size() - 2)?
Extract this as a separate variable?
Is it always 2?